### PR TITLE
automated-comments: Added rust/reverse-string comments

### PR DESCRIPTION
**Pull request by ZapAnton**

### CODE REVIEW

File: automated-comments/rust/reverse-string/solution_function_not_found.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+The test suite for the `reverse-string` exercise expects the following function defined in the `src/lib.rs` file:
+```rust
+pub fn reverse(input: &str) -> String {
+}
+```
+Please, check that a proper function and solution file are submitted.
```
Comment (SleeplessByte) [inline automated-comments/rust/reverse-string/solution_function_not_found.md]: I would remove this line altogether and be more machine-like:

```suggestion
The function was not found
```

File: automated-comments/rust/reverse-string/suggest_removing_extern_crate.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The `extern crate` line [is no longer needed](https://doc.rust-lang.org/stable/edition-guide/rust-2018/module-system/path-clarity.html#no-more-extern-crate) in the 2018 edition of Rust, which the exercises on the Rust track are configured to use.
+Please, consider removing the line.
```
Comment (SleeplessByte) [inline automated-comments/rust/reverse-string/suggest_removing_extern_crate.md]: I would inverse this:

```markdown
Consider removing `extern crate` as it [is no longer needed](https://doc.rust-lang.org/stable/edition-guide/rust-2018/module-system/path-clarity.html#no-more-extern-crate) 
in the 2018 edition of Rust, which the exercises on the Rust track are configured to use.
```

File: automated-comments/rust/reverse-string/suggest_doing_bonus_test.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Consider expanding the solution to pass the feature-gated `test_grapheme_clusters` test case.
+The potential benefits of the expansion is the increased learning in the non-ASCII symbols handling area and the additional training in using the external crates, if the need arises to use one.
```
Comment (SleeplessByte) [inline automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: This is _very_ wordy. Perhaps something simpler?

```suggestion
Understanding how non-ASCII symbols can be handled using external crates has great benefits down the line.
```

Comment (ZapAnton) [reply automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: Yes, you wording is much better.
Just a small moment - there are two items in the comment:

Understanding how non-ASCII symbols can be handled **and**  the usage of the external crates has great benefits down the line.

The student may try to reverse non-ASCII by hand - although daunting, but still a valid approach.

Comment (SleeplessByte) [reply automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: > The student may try to reverse non-ASCII by hand - although daunting, but still a valid approach.

 Hehe, but they probably _shouldn't_ right?

```suggestion
Understanding how non-ASCII symbols can be handled and how to use external crates has great benefits down the line.
```

Like this?

Comment (ZapAnton) [reply automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: Yes, lets leave it at that

File: automated-comments/rust/reverse-string/solution_function_not_found.md (hunk 1 above)
Comment (ZapAnton) [reply automated-comments/rust/reverse-string/solution_function_not_found.md]: Perhaps more explicit:

`The required function was not found`

File: automated-comments/rust/reverse-string/solution_function_not_found.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+The test suite for the `reverse-string` exercise expects the following function defined in the `src/lib.rs` file:
```
Comment (ErikSchierboom) [inline automated-comments/rust/reverse-string/solution_function_not_found.md]:
```suggestion
The test suite expects the following function defined in the `src/lib.rs` file:
```

Maybe mentioning the exercise name is not needed?

File: automated-comments/rust/reverse-string/suggest_doing_bonus_test.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Consider expanding the solution to pass the feature-gated `test_grapheme_clusters` test case.
+Understanding how non-ASCII symbols can be handled and how to use external crates, has great benefits down the line.
```
Comment (ErikSchierboom) [inline automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: Reverse-string is the first exercise after hello-world, right? If so, it might be useful to add some links to pages detailing how to add external crates, and possibly even the crate itself or a hint on what the crate's name is.

File: automated-comments/rust/reverse-string/solution_function_not_found.md (hunk 4 above)
Comment (kotp) [reply automated-comments/rust/reverse-string/solution_function_not_found.md]: I agree that the exercise name is implied via context.  I would commit the suggestion, but no rights to do so.

File: automated-comments/rust/reverse-string/suggest_doing_bonus_test.md (hunk 5 above)
Comment (ZapAnton) [reply automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: The hint on the crate name is already in the README: https://github.com/exercism/rust/blob/master/exercises/reverse-string/README.md#bonus

As for the learning link, how about this:
Understanding how non-ASCII symbols can be handled and how to [use external crates](https://doc.rust-lang.org/cargo/guide/dependencies.html), has great benefits down the line.

File: automated-comments/rust/reverse-string/solution_function_not_found.md (hunk 4 above)
Comment (ZapAnton) [reply automated-comments/rust/reverse-string/solution_function_not_found.md]: How about rewording it to simply `The test suite for this exercise`?

Comment (ErikSchierboom) [reply automated-comments/rust/reverse-string/solution_function_not_found.md]: Sounds good!

File: automated-comments/rust/reverse-string/suggest_doing_bonus_test.md (hunk 5 above)
Comment (ErikSchierboom) [reply automated-comments/rust/reverse-string/suggest_doing_bonus_test.md]: I like the link! And thanks for the crate name explanation.

File: automated-comments/rust/reverse-string/solution_function_not_found.md (hunk 4 above)
Comment (SleeplessByte) [reply automated-comments/rust/reverse-string/solution_function_not_found.md]: Yeah, that's fine.

**Comment by ZapAnton** [comment]
Anything else that needs to be changed before this can be merged?
